report_server/server.py

```diff
@@ -6,56 +7,59 @@
 from bson import ObjectId
 from google.protobuf.empty_pb2 import Empty
 from google.protobuf.json_format import MessageToDict, ParseDict
-from pymongo import MongoClient
+from pymongo import MongoClient, errors
 
-import employee_report_pb2_grpc, employee_report_pb2
+from protocol_buffers import employee_report_pb2_grpc
 
 
-def convert(report):
+def convert(report, message):
     report["id"] = str(report["_id"])
-    return ParseDict(report, employee_report_pb2.Report(), ignore_unknown_fields=True)
+    return ParseDict(report, message, ignore_unknown_fields=True)
 
 
 class EmployeeReportService(employee_report_pb2_grpc.EmployeeReportServiceServicer):
 
     def __init__(self):
-        client = MongoClient('localhost', 27017)
-        self.db = client.service_db
+        try:
+            client = MongoClient('localhost', 27017)
+            self.db = client.service_db
+        except errors.PyMongoError as e:
+            logging.error(str(e))
 
     def CreateReport(self, request, context):
         logging.info('call CreateReport')
         logging.info(request)
         report = MessageToDict(request, preserving_proto_field_name=True)
         self.db.reports.insert_one(report)
-        return convert(report)
+        return convert(report, employee_report_pb2_grpc.Report())
 
     def GetReportById(self, request, context):
         logging.info('call GetReportById')
         logging.info(request)
         report = self.db.reports.find_one({"_id": ObjectId(request.id)})
         if report is not None:
-            return convert(report)
+            return convert(report, employee_report_pb2_grpc.Report())
         else:
             context.set_details('Report not found')
             context.set_code(grpc.StatusCode.NOT_FOUND)
-            return employee_report_pb2.Report()
+            return employee_report_pb2_grpc.Report()
 
     def GetReportsByUser(self, request, context):
         logging.info('call GetReportsByUser')
         logging.info(request)
         reports = self.db.reports.find({"user_id": request.user_id})
-        by_user = employee_report_pb2.ReportsByUser(user_id=request.user_id)
+        by_user = employee_report_pb2_grpc.ReportsByUser(user_id=request.user_id)
         if reports:
-            by_user.reports.extend([convert(report) for report in reports])
+            by_user.reports.extend([convert(report, employee_report_pb2_grpc.Report()) for report in reports])
```

Comment:
by_user.reports = list([convert(report, employee_report_pb2_grpc.Report()) for report in reports])

or 

create a function convert_to_Report with partial function from functools lib, then use map builded_in function

code example:
`from functool import partial

convert_to_Report = partial(convert, message=employee_report_pb2_grpc.Report())

`

so you will be able to use 
`list(map(convert_to_Report, reports))`

https://www.geeksforgeeks.org/python-map-function/
https://www.geeksforgeeks.org/partial-functions-python/

Reply:
why do you think it's better?